subsys/net/ip/tcp2.c

```diff
@@ -951,7 +951,11 @@ static void tcp_in(struct tcp *conn, struct net_pkt *pkt)
 	case TCP_FIN_WAIT_1:
 		if (th && FL(&fl, ==, (FIN | ACK), th_seq(th) == conn->ack)) {
 			tcp_send_timer_cancel(conn);
+			conn_ack(conn, + 1);
+			tcp_out(conn, ACK);
 			next = TCP_TIME_WAIT;
+			k_delayed_work_submit(&conn->timewait_timer,
```

Comment:
@rveerama1 

Subscribing timeout here is an error, it's subscribed in TCP_TIME_WAIT.

Reply:
True, using time_wait_timer is error, but we need a timer here, so that peer receives ACK and we can close the connection gracefully.
May be need to introduce another timer?